os/fs/smartfs/smartfs_smart.c

```diff
@@ -1938,75 +1933,22 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 	oldentry.name = NULL;
 	newentry.name = NULL;
 	ret = smartfs_finddirentry(fs, &oldentry, oldrelpath);
-	if (ret < 0) {
+	if (ret != OK) {
 		fdbg("Old entry doesn't exist\n");
 		goto errout_with_semaphore;
 	}
 
-	/* Search for the new entry and validate it DOESN'T exist, unless we
-	 * are copying to a directory and keeping the same file name, such as:
-	 *
-	 *    mv /mntpoint/somefile.txt /mntpoint/somedir
-	 *
-	 * in which case, we need to validate the /mntpoint/somedir/somefile.txt
-	 * doens't exsit.
-	 */
-
+	/* Search for the new entry and validate it DOESN'T exist */
 	ret = smartfs_finddirentry(fs, &newentry, newrelpath);
 	if (ret == OK) {
-		/* Test if it's a file.  If it is, then it's an error */
-
-		if ((newentry.flags & SMARTFS_DIRENT_TYPE) == SMARTFS_DIRENT_TYPE_FILE) {
-			ret = -EEXIST;
+		ret = -EEXIST;
+		fdbg("Entry of the same name already exists, cannot rename\n");
+		goto errout_with_semaphore;
+	} else {
+		if (ret != -ENOENT) {
+			fdbg("Cannot rename entry\n");
```

Comment:
Here we could update error number in log

Reply:
Smartfs does not always print the error code on the console, keeping the same for now.